lib/ui/form/widget/form_survey_question_page.dart

```diff
@@ -66,25 +69,71 @@ class FormSurveyQuestionPage extends StatelessWidget {
     final displayType = question.displayType;
     switch (displayType) {
       case DisplayType.smiley:
-        return const FormSurveyAnswerSmiley();
+        return FormSurveyAnswerSmiley(
+          answers: question.answers,
+          onUpdateAnswer: (selectedAnswer) {
+            onUpdatedAnswers([selectedAnswer]);
+          },
+        );
       case DisplayType.nps:
         return FormSurveyAnswerNps(
-          question: question,
+          answers: question.answers,
+          onUpdateAnswer: (selectedAnswer) {
+            onUpdatedAnswers([selectedAnswer]);
+          },
         );
       case DisplayType.heart:
-        return FormSurveyAnswerEmoji(emoji: '❤');
+        return FormSurveyAnswerEmoji(
+          emoji: '❤',
+          answers: question.answers,
+          onUpdateAnswer: (selectedAnswer) {
+            onUpdatedAnswers([selectedAnswer]);
+          },
+        );
       case DisplayType.star:
-        return FormSurveyAnswerEmoji(emoji: '⭐');
+        return FormSurveyAnswerEmoji(
+          emoji: '⭐',
+          answers: question.answers,
+          onUpdateAnswer: (selectedAnswer) {
+            onUpdatedAnswers([selectedAnswer]);
+          },
+        );
       case DisplayType.thumbs:
-        return FormSurveyAnswerEmoji(emoji: '👍🏻');
+        return FormSurveyAnswerEmoji(
+          emoji: '👍🏻',
+          answers: question.answers,
+          onUpdateAnswer: (selectedAnswer) {
+            onUpdatedAnswers([selectedAnswer]);
+          },
+        );
       case DisplayType.textarea:
-        return const FormSurveyAnswerTextarea();
+        return FormSurveyAnswerTextarea(
+          answers: question.answers,
+          onUpdateText: (answer) {
+            onUpdatedAnswers([answer]);
+          },
+        );
```

Comment:
Should we check the list of `answers` is not empty before rendering the widget? Getting `answers.first` inside the widget is not safe.

```suggestion
        if (question.answers.isNotEmpty) {
          return FormSurveyAnswerTextarea(
            answer: question.answers.first,
            onUpdateText: (answer) {
              onUpdatedAnswers([answer]);
            },
          );
        } else {
          return const SizedBox.shrink();
        }
```